### PR TITLE
Sketcher: Modernize SketchObject.cpp

### PR DESCRIPTION
**Pull request by PaddleStroke**
While going to fix a build error of GeomAPI_ProjectPointOnCurve (see other PR), I couldn't help myself and modernized some stuff  in SketchObject.cpp : 
- for loops
- some auto
- add some {} to ifs.

### CODE REVIEW

**Comment by chennes** [comment]
Can you please rebase? That will clean up most of the CI failures (I think Windows non-Conda will still report a test failure that I'm looking into).

**Comment by PaddleStroke** [comment]
@chennes CI fixed after rebase.

File: src/Mod/Sketcher/App/SketchObject.cpp [hunk 1]
```diff
@@ -6196,28 +6198,30 @@ int SketchObject::deleteUnusedInternalGeometry(int GeoId, bool delgeoid)
...
+            else if (constr->Second == focus1elementindex || constr->First == focus1elementindex
+                     || constr->Third == focus1elementindex)
                 focus1constraints++;
         }
 
         std::vector<int> delgeometries;
 
         // major has minimum one constraint, the specific internal alignment constraint
-        if (majorelementindex != -1 && majorconstraints < 2)
+        if (majorelementindex != -1 && majorconstraints < 2){
```
[the remark this answers is not in the document]
Comment (PaddleStroke) [reply src/Mod/Sketcher/App/SketchObject.cpp]: space added.

File: src/Mod/Sketcher/App/SketchObject.cpp [hunk 2]
```diff
@@ -5785,13 +5784,17 @@ int SketchObject::exposeInternalGeometry(int GeoId)
         this->addGeometry(igeo, true);
         this->addConstraints(icon);
 
-        for (std::vector<Part::Geometry*>::iterator it = igeo.begin(); it != igeo.end(); ++it)
-            if (*it)
-                delete *it;
+        for (auto* geo : igeo) {
+            if (geo) {
+                delete geo;
+            }
```
Comment (AjinkyaDahale) [inline src/Mod/Sketcher/App/SketchObject.cpp:5790]: `clang-tidy` suggests this check can be removed. Deleting a `nullptr` does nothing. That said it's not a major concern.

**Review by AjinkyaDahale — APPROVED** [review]

File: src/Mod/Sketcher/App/SketchObject.cpp (hunk 2 above)
Comment (PaddleStroke) [reply src/Mod/Sketcher/App/SketchObject.cpp:5790]: Why do you say that 'Deleting a nullptr does nothing' ? This is deleting geo if it is not nullptr.

Besides this PR is just simplifying the syntax. If this needs to be removed it should be a separate PR.

Comment (AjinkyaDahale) [reply src/Mod/Sketcher/App/SketchObject.cpp:5790]: I mean this can be replaced with:
```cpp
for (auto geo : ...) {
    delete geo;
}
```

**Comment by PaddleStroke** [comment]
@AjinkyaDahale I have removed the unnecessary if(geo) please check again.
Thanks

File: src/Mod/Sketcher/App/SketchObject.cpp [hunk 3]
```diff
@@ -6413,8 +6398,9 @@ int SketchObject::deleteUnusedInternalGeometry(int GeoId, bool delgeoid)
         }
 
 
-        if (delgeoid)
+        if (delgeoid){
```
Comment (AjinkyaDahale) [inline src/Mod/Sketcher/App/SketchObject.cpp:6401]: Still a missing space.

**Review by AjinkyaDahale — APPROVED** [review]

**Comment by maxwxyz** [comment]
@PaddleStroke could you please resolve the conflicts? TIA

**Comment by PaddleStroke** [comment]
I will check this but it seems there're a lot of conflict so I may just drop it. I can feel this is just going to cause a lot of annoying issues/rebasing.

**Comment by PaddleStroke** [comment]
Closing this is going to be a mess/redundant with @AjinkyaDahale's work.

**Pull request closed without merging**